.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - feature-branch
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker Hub
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+
+      - name: Build Docker Image
+        run: |
+          docker buildx build --file Dockerfile -t ${{ secrets.DOCKER_USERNAME }}/homework-docker2:tagname .
+          docker push ${{ secrets.DOCKER_USERNAME }}/homework-docker2:tagname
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+
+      - name: Logout from Docker Hub
+        run: docker logout
```

Comment:
This is redundant. Nothing left after the build is done unless you explicitly cache it.